src/networking.c

```diff
@@ -43,6 +43,7 @@ static void pauseClientsByClient(mstime_t end, int isPauseClientAll);
 int postponeClientRead(client *c);
 char *getClientSockname(client *c);
 int ProcessingEventsWhileBlocked = 0; /* See processEventsWhileBlocked(). */
+__thread sds shared_qb = NULL;
```

Comment:
Is `__thread` a compiler-specific extension? Which C standard is this from? I found this: https://gcc.gnu.org/onlinedocs/gcc/Thread-Local.html

We should at least comment about it here, since we support a lot of different systems.

I think there are other keywords without leading underscore, and `<thread.h>`, but I'm not sure which C versions this stuff was introduced.

Reply:
I see now that `thread_local` was added in C11. The reason we still use C99 is that CentOS 7 has an old compiler that doesn't support C11. (See 445a4b669a3a7232a18bf23340c5f7d580aa92c7)

That decision was made some years ago and CentOS will be end of life in June this year. That's before our next release, so I think we can start using C11 fully. @valkey-io/core-team WDYT?

Reply:
>I think we can start using C11 fully

I am aligned with this being the default stance.

Reply:
> That decision was made some years ago and CentOS will be end of life in June this year. That's before our next release, so I think we can start using C11 fully.

My stance is let's do the same thing Redis did before, which is let's update it and see if folks complain and then revert it if they do.

Reply:
Are we pulling this to a follow up PR then?

Reply:
Sure, we can do the switch to C11 in a follow-up PR, so it's easy to revert if we must. (Possibly we can get rid of atomicvar.h too and use C11 atomics directly everywhere.)

Reply:
Yeah, that sounded right to me.